### PR TITLE
enable transparent hugepage for kernel-mshv

### PR DESCRIPTION
**Pull request by Redent0r**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
This PR is for https://microsoft.visualstudio.com/OS/_workitems/edit/41985970/. There's an AKS E2E test that consistently fails on Kata images. It particularly fails trying to assess the existence of `/sys/kernel/mm/transparent_hugepage/enabled` and `/sys/kernel/mm/transparent_hugepage/defrag` (https://www.kernel.org/doc/html/latest/admin-guide/mm/transhuge.html). Kata images use kernel-mshv. Turning this options in the kernel-mshv spec gets this file created and will make the test pass for Kata image.

###### Change Log  

- kernel-mshv

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- https://microsoft.visualstudio.com/OS/_workitems/edit/41985970/

###### Test Methodology

- Updated config and built kernel-mshv package locally. Then installed on test node running Kata. And I was able to verify the necessary files get created as expected.

Before: transparent hugepage files are missing
Screenshot 2022-11-07 112757

After: transparent hugepage files are present as expected
Screenshot 2022-11-10 134914

### CODE REVIEW

File: SPECS-EXTENDED/kernel-mshv/config [hunk 1]
```diff
@@ -939,7 +939,9 @@ CONFIG_DEFAULT_MMAP_MIN_ADDR=65536
 CONFIG_ARCH_SUPPORTS_MEMORY_FAILURE=y
 CONFIG_MEMORY_FAILURE=y
 # CONFIG_HWPOISON_INJECT is not set
-# CONFIG_TRANSPARENT_HUGEPAGE is not set
+CONFIG_TRANSPARENT_HUGEPAGE=y
```
Comment (Redent0r) [inline SPECS-EXTENDED/kernel-mshv/config:942]: Mirroring [kernel config](https://github.com/microsoft/CBL-Mariner/blob/main/SPECS/kernel/config#L980)

**Review by neha170 — COMMENTED** [review]
Thanks for making the change, Saul. The change itself looks good to me. I am only wondering how we have identified if we need to pass this test or not, and if this might adversely affect the performance. I think it would be nice to weigh where we stand w.r.t performance v/s tests passed.
Kindly also make the target branch 'main'.

**Comment by Redent0r** [comment]
Thanks Neha! This makes sense to me; I'll make a bit more research and weigh in test coverage vs performance impact.

**Review by christopherco — APPROVED** [review]

**Comment by Redent0r** [comment]
I reached out to Wei and doesn't look like there are any perceivable performance implications. What do you think @neha170 ?
image

**Comment by Redent0r** [comment]
Hold! I'm double checking the e2e test. It was failing on the cat `/sys/kernel/mm/transparent_hugepage/enabled` and `/sys/kernel/mm/transparent_hugepage/defrag` [here](https://msazure.visualstudio.com/CloudNativeCompute/_git/aks-rp?path=/test/e2e/suites/customnodeconfig/main.go&version=GBmaster&line=184&lineEnd=185&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents). I thought just creating the file was enough. But it actually expects a specific file content below [here](https://msazure.visualstudio.com/CloudNativeCompute/_git/aks-rp?path=/test/e2e/suites/customnodeconfig/main.go&version=GBmaster&line=290&lineEnd=291&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents). Silly me I just did an `ls` in the summary screenshot and I'm not sure what's the config in the file. I need to double check this config will create the expected file content. Will fix it and add screenshot confirming it produces the expected config

**Comment by Redent0r** [comment]
Here we go. This config produces `[always] madvise never` for `enabled` and `always defer defer+madvise [madvise] never` for `defrag`
image

But according to the [test assesement](https://msazure.visualstudio.com/CloudNativeCompute/_git/aks-rp?path=/test/e2e/suites/customnodeconfig/main.go&version=GBmaster&line=290&lineEnd=291&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents), we need `always [madvise] never` for `enabled` and `always defer [defer+madvise] madvise never` for defrag. 

I'll need to tweak the config a bit more and make them match with what the test expects. But we are in right direction

**Comment by Redent0r** [comment]
[This](https://github.com/microsoft/CBL-Mariner/pull/4201/commits/5f6b19895d5beb99adf717d81774f8744cd0036a) fixed `enabled`. Just need to fix `defrag`
image

**Comment by Redent0r** [comment]
I just realized that the expectations of `enabled` and `defrag` content are [set by the test itself](https://msazure.visualstudio.com/CloudNativeCompute/_git/aks-rp?path=/test/e2e/suites/customnodeconfig/main.go&version=GBmaster&line=134&lineEnd=134&lineStartColumn=2&lineEndColumn=53&lineStyle=plain&_a=contents). I confirmed this by changing the test expectation of defrag (from `defer+madvise` to `defer`) and seeing how the `defrag` content [gets updated accordingly on marinerv2](https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=65418359&view=logs&j=f4e8c341-b016-5e25-d328-6bb0b10d81c8&t=8aa97eef-2a7a-5d58-7169-cc08b59be4aa&l=3275). So, it should be good with just mirroring kernel config from mariner and making sure that those files are created. The test itself will update the content of those files and it should pass

**Review by neha170 — APPROVED** [review]
Thanks for the extensive testing and diving in, Saul

**Pull request merged**